### ISSUE TITLE
parallel_crand_ ValueError

### ISSUE THREAD

**Issue by hh2110** [pysal/esda#146]
Hi everyone. First of all major kudos to the developers of this library - it is really cool!

I’m trying to take advantage of the parallelisation opportunity when calculating p values in the calculation of moran’s local index. 

Here is my simple code

```
import esda
import numpy as np
import pysal as ps
import time
time0 = time.time()

size = 100;
patch_eg = np.random.rand(size,size)
w = ps.lib.weights.lat2W(size,size)
lm = esda.Moran_Local(patch_eg, w, transformation='r', permutations=10, n_jobs=8)
moran_local = np.reshape(lm.Is, patch_eg.shape); print(moran_local)
print('time take', time.time() - time0)
```

And the error I get when running this is
```

Traceback (most recent call last):
  File "try.py", line 11, in <module>
    lm = esda.Moran_Local(patch_eg, w, transformation='r', permutations=10, n_jobs=8)
  File "/home/hh774582/.conda/envs/fam13a-dev/lib/python3.7/site-packages/esda/moran.py", line 1017, in __init__
    seed=seed,
  File "/home/hh774582/.conda/envs/fam13a-dev/lib/python3.7/site-packages/esda/crand.py", line 184, in crand
    stat_func,
  File "/home/hh774582/.conda/envs/fam13a-dev/lib/python3.7/site-packages/esda/crand.py", line 467, in parallel_crand
    rlocals = np.hstack(rlocals).flatten()
  File "<__array_function__ internals>", line 6, in hstack
  File "/home/hh774582/.conda/envs/fam13a-dev/lib/python3.7/site-packages/numpy/core/shape_base.py", line 345, in hstack
    return _nx.concatenate(arrs, 1)
  File "<__array_function__ internals>", line 6, in concatenate
ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 0, the array at index 0 has size 1251 and the array at index 7 has size 1243
```

I thought this is related to numba so I have updated to 0.50.1 but still get the above error. Any help is greatly appreciated. Thanks.

**Comment by ljwolf** [comment]
Thanks for the report! Can you please double check that `import numba; numba.__version__` is correct? When developing this, we occasionally had issues with ensuring that the versions were correct. 

Second, I note that you have only 10 permutations. How many observations is in `patch_eg`?

**Comment by hh2110** [comment]
Hi, I have printed the numba version and it is 0.50.1 - `patch_eg` is 100*100 observations.

**Comment by ljwolf** [comment]
Great, thanks. 

Have you reshaped `patch_eg` to be flat?

**Comment by hh2110** [comment]
Yes, no luck there either. I have tried 2D arrays before with n_jobs=1 and it does work for me.

**Comment by ljwolf** [comment]
hmm, OK. Can you post the data?

**Comment by hh2110** [comment]
Do you mean patch_eg? It is just np.random.rand(size,size) - so just a 2D matrix of random numbers. I am using size=20 for now

**Comment by ljwolf** [comment]
Ah, Yes, Sorry! I guess I missed that :smile: 

I can definitely replicate this. It appears to be an issue I thought we addressed in development. I'll take a look & see if we can't get a bug fix.

**Comment by hh2110** [comment]
Thank you!

**Comment by ljwolf** [comment]
I think an immediate workaround is `keep_simulation=False`? The issue appears to be that the `rlocals` array is not concatenating correctly.

**Comment by hh2110** [comment]
Got it, thanks - it does work plus I don’t really need the simulations. Than you for your help.

**Comment by ljwolf** [comment]
Could you give the output of your `conda list`?

**Comment by hh2110** [comment]
yes sure, it is attached
[conda_list.txt](https://github.com/pysal/esda/files/5117081/conda_list.txt)

**Comment by ljwolf** [comment]
Thanks! That's all. I have a tentative fix pending.  Thank you for the report!

**Comment by sjsrey** [comment]
Resolved with #147
